src/components/Workouts/Workouts.js

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useState } from 'react';
+import Stopwatch from "../Stopwatch/Stopwatch";
+import getTimeFormated from '../../utils/getTimeFormated'
+import { format } from 'date-fns'
+
+import './Workouts.scss';
+
+export default function Workout() {
+    const [categories, setCategories] = useState([]);
+    const [selectedCategory, setSelectedCategory] = useState();
+    const [workouts, setWorkouts] = useState([]);
+    const [selectedWorkout, setSelectedWorkout] = useState(null);
+    const [trainings, setTrainings] = useState([]);
+
+    useEffect(() => {
+        const fetchData = () => {
+            fetch('http://localhost:3001/category')
+                .then((response) => {
+                    if (!response.ok) {
+                        throw new Error('Network response was not ok');
+                    }
+                    return response.json();
+                })
+                .then((data) => {
+                    setCategories(data);
+                })
+                .catch(error => {
+                    console.error('Error:', error);
+                });
+        };
+        fetchData();
+    }, [])
+
+    useEffect(() => {
+        let selectedCategoryId = (selectedCategory == null) ? "" : selectedCategory.id;
+        const fetchData = () => {
+            fetch(`http://localhost:3001/workouts?categoryId=${selectedCategoryId}`)
+                .then((response) => {
+                    if (!response.ok) {
+                        throw new Error('Network response was not ok');
+                    }
+                    return response.json();
+                })
+                .then((data) => {
+                    setWorkouts(data);
+                })
+                .catch(error => {
+                    console.error('Error:', error);
+                });
+        };
+        fetchData();
+    }, [selectedCategory])
+
+    useEffect(() => {
```

Comment:
approach no 1: extract to separate function
const getAsync = (url) => {
fetch(url)
                .then((response) => {
                    if (!response.ok) {
                        throw new Error('Network response was not ok');
                    }
                    return response.json();
                })
                .then((data) => {
                    setTrainings(data);
                })
                .catch(error => {
                    console.error('Error:', error);
                });
}

or 
    useEffect(() => {
const getAsync = async (url) => {
const response = await fetch(url);
if (!response.ok) {
    throw new Error('Network response was not ok');
}
return await response.json();
}

or hook 
const useFetch = (url) => {
const [data, setData] = useState([]);
const [status, setStatus] = useState(requestStatus.unknown);

useEffect(() => {
  getAsync(url);
}, [url]);

const getAsync = useCallback((url) => {
   
   fetch(url)
                .then((response) => {
                    if (!response.ok) {
                        setStatus(requestStatus.error);
                        throw new Error('Network response was not ok');
                    }
                    return response.json();
                })
                .then((data) => {
                    setTrainings(data);
                        setStatus(requestStatus.success);
                })
                .catch(error => {
                       setStatus(requestStatus.error);
                    console.error('Error:', error);
                });
  }
}, []);
return {data, status }



const fetchTrainings = async () => {
const trainingsData = await getAsync(..../training);
setTrainings(data);



useEffect(() => {
const fetchTrainings = async () => {
   const trainingsData = await getAsync(..../training);
    setTrainings(data);
    }
    fetchTrainings();
}, [])

approach no 2: use hook

const requestStatus =  {
 unknown = "unknown",
 isLoading = "isLoading",
 ...
}

const useTrainings = (url) => {
 const {status, data} = useFetch(url);
 return {data, status}
}


usage in component:
const {trainings, status: trainingsStatus} = useTrainings();